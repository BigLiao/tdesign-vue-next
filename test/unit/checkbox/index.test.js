import { mount } from '@vue/test-utils';
import Checkbox, { Group } from '@/src/checkbox/index.ts';

// every component needs four parts: props/events/slots/functions.
describe('Checkbox', () => {
  // test props api
  describe(':props', () => {
    it(':checked', () => {
      const wrapper = mount({
        render() {
          return <Checkbox checked={true}></Checkbox>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':defaultChecked', () => {
      const wrapper = mount({
        render() {
          return <Checkbox defaultChecked={true}></Checkbox>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':disabled', () => {
      const wrapper = mount({
        render() {
          return <Checkbox disabled={true}></Checkbox>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':indeterminate', () => {
      const wrapper = mount({
        render() {
          return <Checkbox indeterminate={true}></Checkbox>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':name', () => {
      const wrapper = mount({
        render() {
          return <Checkbox name={'checkbox-name'}></Checkbox>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  // test events
  describe('@event', () => {
    it('Event passthrough ', () => {
      const fn = jest.fn();
      const wrapper = mount({
        render() {
          return <Checkbox onChange={fn}>Checkbox</Checkbox>;
        },
      });
      wrapper.find(Checkbox).trigger('click');
      expect(fn).toHaveBeenCalled();;
    });
  });
});

describe('Checkbox Group', () => {
  // test props api
  describe(':props', () => {
    it(':defaultValue', () => {
      const wrapper = mount({
        render() {
          return (
            <Group checked={['sz']}>
              <Checkbox value="gz">广州</Checkbox>
              <Checkbox value="sz" disabled>深圳</Checkbox>
            </Group>
          );
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':value', () => {
      const wrapper = mount({
        render() {
          return (
            <Group value={['sz']}>
              <Checkbox value="gz">广州</Checkbox>
              <Checkbox value="sz" disabled>深圳</Checkbox>
            </Group>
          );
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':disabled', () => {
      const wrapper = mount({
        render() {
          return (
            <Group disabled={true}>
              <Checkbox value="gz">广州</Checkbox>
              <Checkbox value="sz" disabled>深圳</Checkbox>
            </Group>
          );
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':options', () => {
      const options = [
        { value: 'gz', label: '广州', disabled: true },
        { value: 'sz', label: '深圳' },
      ];
      const wrapper = mount({
        render() {
          return <Group options={options}></Group>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
    it(':name', () => {
      const wrapper = mount({
        render() {
          return (
            <Group name={'checkbox-name'}>
              <Checkbox value="gz">广州</Checkbox>
              <Checkbox value="sz" disabled>深圳</Checkbox>
            </Group>
          );
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  // test events
  describe('@event', () => {
    it('Event passthrough ', () => {
      const fn = jest.fn();
      const wrapper = mount({
        render() {
          return (
            <Group onChange={fn}>
              <Checkbox value="gz">广州</Checkbox>
              <Checkbox value="sz" disabled>深圳</Checkbox>
            </Group>
          );
        },
      });
      wrapper.find(Checkbox).trigger('click');
      expect(fn).toHaveBeenCalled();;
    });
  });
});
