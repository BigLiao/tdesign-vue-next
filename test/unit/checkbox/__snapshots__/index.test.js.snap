// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox :props :checked 1`] = `<label class="t-checkbox t-is-checked"><input type="checkbox" class="t-checkbox__former" value=""><span class="t-checkbox__input"></span><span class="t-checkbox__label"></span></label>`;

exports[`Checkbox :props :defaultChecked 1`] = `<label class="t-checkbox t-is-checked"><input type="checkbox" class="t-checkbox__former" value=""><span class="t-checkbox__input"></span><span class="t-checkbox__label"></span></label>`;

exports[`Checkbox :props :disabled 1`] = `<label class="t-checkbox t-is-disabled"><input type="checkbox" class="t-checkbox__former" disabled="" value=""><span class="t-checkbox__input"></span><span class="t-checkbox__label"></span></label>`;

exports[`Checkbox :props :indeterminate 1`] = `<label class="t-checkbox t-is-indeterminate"><input type="checkbox" class="t-checkbox__former" value=""><span class="t-checkbox__input"></span><span class="t-checkbox__label"></span></label>`;

exports[`Checkbox :props :name 1`] = `<label class="t-checkbox"><input type="checkbox" class="t-checkbox__former" value="" name="checkbox-name"><span class="t-checkbox__input"></span><span class="t-checkbox__label"></span></label>`;

exports[`Checkbox Group :props :defaultValue 1`] = `<div class="t-checkbox-group"><label class="t-checkbox"><input type="checkbox" class="t-checkbox__former" value="gz"><span class="t-checkbox__input"></span><span class="t-checkbox__label">广州</span></label><label class="t-checkbox t-is-checked t-is-disabled"><input type="checkbox" class="t-checkbox__former" disabled="" value="sz"><span class="t-checkbox__input"></span><span class="t-checkbox__label">深圳</span></label></div>`;

exports[`Checkbox Group :props :disabled 1`] = `<div class="t-checkbox-group"><label class="t-checkbox"><input type="checkbox" class="t-checkbox__former" value="bj"><span class="t-checkbox__input"></span><span class="t-checkbox__label">北京</span></label><label class="t-checkbox"><input type="checkbox" class="t-checkbox__former" value="gz"><span class="t-checkbox__input"></span><span class="t-checkbox__label">广州</span></label><label class="t-checkbox"><input type="checkbox" class="t-checkbox__former" value="sz"><span class="t-checkbox__input"></span><span class="t-checkbox__label">深圳</span></label></div>`;

exports[`Checkbox Group :props :name 1`] = `<div class="t-checkbox-group"><label class="t-checkbox"><input type="checkbox" class="t-checkbox__former" value="gz" name="checkbox-name"><span class="t-checkbox__input"></span><span class="t-checkbox__label">广州</span></label><label class="t-checkbox t-is-disabled"><input type="checkbox" class="t-checkbox__former" disabled="" value="sz" name="checkbox-name"><span class="t-checkbox__input"></span><span class="t-checkbox__label">深圳</span></label></div>`;

exports[`Checkbox Group :props :options 1`] = `<div class="t-checkbox-group"><label class="t-checkbox t-is-disabled"><input type="checkbox" class="t-checkbox__former" disabled="" value="gz"><span class="t-checkbox__input"></span><span class="t-checkbox__label">广州</span></label><label class="t-checkbox"><input type="checkbox" class="t-checkbox__former" value="sz"><span class="t-checkbox__input"></span><span class="t-checkbox__label">深圳</span></label></div>`;

exports[`Checkbox Group :props :value 1`] = `<div class="t-checkbox-group"><label class="t-checkbox"><input type="checkbox" class="t-checkbox__former" value="gz"><span class="t-checkbox__input"></span><span class="t-checkbox__label">广州</span></label><label class="t-checkbox t-is-checked t-is-disabled"><input type="checkbox" class="t-checkbox__former" disabled="" value="sz"><span class="t-checkbox__input"></span><span class="t-checkbox__label">深圳</span></label></div>`;
